web_pwa_oca/controllers/main.py

```diff
@@ -14,53 +26,58 @@ def get_asset_urls(self, asset_xml_id):
+        return [
+            cache_version,
+            urls
+        ]
+
+    @route('/service-worker.js', type='http', auth="public")
+    def render_service_worker(self):
+        """ Route to register the service worker in the 'main' scope ('/') """
+        return request.render('web_pwa_oca.service_worker', {
+            'pwa_scripts': self._get_pwa_scripts(),
+            'pwa_params': self._get_pwa_params(),
+        }, headers=[('Content-Type', 'text/javascript;charset=utf-8')])
 
     @route('/web_pwa_oca/manifest.json', type='http', auth="public")
-    def manifest(self):
-        qweb = request.env['ir.qweb'].sudo()
-        config_param = request.env['ir.config_parameter'].sudo()
-        pwa_name = config_param.get_param("pwa.manifest.name", "Odoo PWA")
-        pwa_short_name = config_param.get_param("pwa.manifest.short_name", "Odoo PWA")
-        icon128x128 = config_param.get_param(
+    def render_manifest(self):
+        """ Render the manifest used to install the page as app """
+        config_param_sudo = request.env['ir.config_parameter'].sudo()
+        pwa_name = config_param_sudo.get_param("pwa.manifest.name", "Odoo PWA")
+        pwa_short_name = config_param_sudo.get_param(
+            "pwa.manifest.short_name", "Odoo PWA")
+        icon128x128 = config_param_sudo.get_param(
             "pwa.manifest.icon128x128",
             "/web_pwa_oca/static/img/icons/icon-128x128.png")
-        icon144x144 = config_param.get_param(
+        icon144x144 = config_param_sudo.get_param(
             "pwa.manifest.icon144x144",
             "/web_pwa_oca/static/img/icons/icon-144x144.png")
-        icon152x152 = config_param.get_param(
+        icon152x152 = config_param_sudo.get_param(
             "pwa.manifest.icon152x152",
             "/web_pwa_oca/static/img/icons/icon-152x152.png")
-        icon192x192 = config_param.get_param(
+        icon192x192 = config_param_sudo.get_param(
             "pwa.manifest.icon192x192",
             "/web_pwa_oca/static/img/icons/icon-192x192.png")
-        icon256x256 = config_param.get_param(
+        icon256x256 = config_param_sudo.get_param(
             "pwa.manifest.icon256x256",
             "/web_pwa_oca/static/img/icons/icon-256x256.png")
-        icon512x512 = config_param.get_param(
+        icon512x512 = config_param_sudo.get_param(
             "pwa.manifest.icon512x512",
             "/web_pwa_oca/static/img/icons/icon-512x512.png")
-        background_color = config_param.get_param(
+        background_color = config_param_sudo.get_param(
             "pwa.manifest.background_color", "#2E69B5")
-        theme_color = config_param.get_param(
+        theme_color = config_param_sudo.get_param(
             "pwa.manifest.theme_color", "#2E69B5")
-        mimetype = 'application/json;charset=utf-8'
-        content = qweb.render('web_pwa_oca.manifest', {
+        return request.render('web_pwa_oca.manifest', {
```

Comment:
This should return the JSON dict directly. Using a template to render a JSON is a bad idea.